src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -729,21 +802,102 @@ def select_units(self, unit_ids, format="memory", folder=None) -> "SortingAnalyz
+
+        Parameters
+        ----------
+        units_to_merge : list/tuple of lists/tuples
+            A list of lists for every merge group. Each element needs to have at least two elements (two units to merge),
+            but it can also have more (merge multiple units at once).
+        new_unit_ids : None or list
+            A new unit_ids for merged units. If given, it needs to have the same length as `units_to_merge`. If None,
+            merged units will have the first unit_id of every lists of merges
+        censor_ms : None or float
+            When merging units, any spikes violating this refractory period will be discarded. Default is None
+        merging_mode : "soft" can be in ["soft", "hard"]
+            How merges are performed. In the "soft" mode, merges will be approximated, with no reloading of the
+            waveforms. This will lead to approximations. If "hard", recomputations are accuratly performed,
+            reloading waveforms if needed
+        sparsity_overlap : float, default 0.75
+            The percentage of overlap that units should share in order to accept merges. If this criteria is not
+            achieved, soft merging will not be possible
+        new_id_strategy : "append" | "take_first", default: "append"
+            The strategy that should be used, if `new_unit_ids` is None, to create new unit_ids.
+                * "append" : new_units_ids will be added at the end of max(sorging.unit_ids)
+                * "take_first" : new_unit_ids will be the first unit_id of every list of merges
+        folder : Path or None
+            The new folder where selected waveforms are copied
+        format : "auto" | "binary_folder" | "zarr"
+            The format of the folder.
+        verbose:
+
+
+        Returns
+        -------
+        analyzer :  SortingAnalyzer
+            The newly create sorting_analyzer with the selected units
+        """
+
+        assert merging_mode in ["soft", "hard"], "Merging mode should be either soft or hard"
+
+        if len(units_to_merge) == 0:
+            return self
+
+        for units in units_to_merge:
+            if len(units) < 2:
+                raise ValueError("Merging requires at least two units to merge")
+
+        if not isinstance(units_to_merge[0], (list, tuple)):
+            # keep backward compatibility : the previous behavior was only one merge
+            units_to_merge = [units_to_merge]
+
+        new_unit_ids = generate_unit_ids_for_merge_group(self.unit_ids, units_to_merge, new_unit_ids, new_id_strategy)
+
+        return self._save_or_select_or_merge(
+            format=format,
+            folder=folder,
+            units_to_merge=units_to_merge,
+            unit_ids=new_unit_ids,
+            censor_ms=censor_ms,
+            merging_mode=merging_mode,
+            sparsity_overlap=sparsity_overlap,
+            verbose=verbose,
+            **job_kwargs,
```

Comment:
new_id_strategy need to be propagate no ?

Reply:
new_id_strategy is handled at the merge_units() level, and then, after, new_units_id is never None